src/Emulator/Peripherals/Peripherals/MTD/OpenTitan_FlashController.cs

```diff
@@ -247,14 +255,31 @@ public OpenTitan_FlashController(Machine machine, MappedMemory flash) : base(mac
                 .WithReservedBits(5, 27);
 
             Registers.ErrorCode.Define(this)
-                .WithFlag(0, out outOfBoundsError, FieldMode.Read | FieldMode.WriteOneToClear, name: "oob_err")
+                .WithFlag(0, out outOfBoundsError, FieldMode.Read | FieldMode.WriteOneToClear, name: "op_err")
+                .WithFlag(1, out memoryProtectionError, FieldMode.Read | FieldMode.WriteOneToClear, name: "mp_err")
+                .WithTaggedFlag("rd_err", 2)
+                .WithTaggedFlag("prog_err", 3)
+                .WithTaggedFlag("prog_win_err", 4)
+                .WithTaggedFlag("prog_type_err", 5)
+                .WithTaggedFlag("flash_macro", 6)
+                .WithTaggedFlag("update_err", 7)
+                .WithReservedBits(8, 24);
+
+            Registers.FaultStatus.Define(this)
+                .WithFlag(0, out outOfBoundsError, FieldMode.Read | FieldMode.WriteOneToClear, name: "op_err")
```

Comment:
This overwrites the field set when generating the `ErrorCode` register.

Reply:
? `FaultStatus` is at 0x178, whereas `ErrorCode` is 0x174. They should be independent controls. Without defining this register, you will see a large spamming of warning messages when running flash_ctrl smoketest

```
16:33:18.1095 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1097 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1137 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1138 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1164 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1164 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1182 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1183 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1185 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1185 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1188 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
16:33:18.1188 [WARNING] flash_ctrl: Unhandled write to offset 0x178, value 0x0.
16:33:18.1189 [WARNING] flash_ctrl: Unhandled read from offset 0x178.
...
```

Reply:
What I meant was that two registers use `out outOfBoundError`, so this C# class field is first set when the `ErrorCode` register is being created and than is being overwritten by `FaultStatus`.

This could be resolved by simply using a separate field (a C# field in the class) for `FaultStatus.op_err`, but if I understand correctly it's not needed. Since we just want to hush warnings about accessing register at offset 0x178, it should be enough to use just tags in `FaultStatus`.